packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,364 @@
+	editProp: EditState | undefined;
+	setEditProp: React.Dispatch<React.SetStateAction<EditState | undefined>>;
+	submitChange: (data: EditData) => void;
+}
+
+interface EditState {
+	type: string;
+	value?: EditData;
+}
+/**
+ * Render data that is Editable with its corresponding UI
+ *
+ * @remarks {@link MessageRelayContext} must be set in order to use this component.
+ */
+export function EditableView(props: EditableViewProps): React.ReactElement {
+	const { node, containerKey, label } = props;
+
+	const [editProp, setEditProp] = React.useState<EditState | undefined>(undefined);
+
+	const messageRelay = useMessageRelay();
+
+	/**
+	 * Using the {@link fluidObjectId}, {@link containerKey}, and data posts a message to edit the DDS
+	 * @param data - The data to edit the DDS with
+	 */
+	const submitChange = React.useCallback(
+		(data: EditData): void => {
+			const edit = {
+				fluidObjectId: node.fluidObjectId,
+				data,
+			};
+
+			messageRelay.postMessage(
+				DataEdit.createMessage({
+					containerKey,
+					edit,
+				}),
+			);
+			setEditProp(undefined);
+		},
+		[containerKey, messageRelay, node.fluidObjectId],
+	);
+
+	/**
+	 * On blur will set {@link isEditing} to false
+	 * Blurring should occur when the user discards an edit (Componenet specific)
+	 * or focuses something not part of the editing experience of that node
+	 */
+	function onBlur(event: React.FocusEvent<HTMLDivElement>): void {
+		/**
+		 * This allows the parent element to share focus with children
+		 * so the blur event only triggers if focus is outside of div and children
+		 */
+		if (!event.currentTarget.contains(event.relatedTarget)) {
+			setEditProp(undefined);
+		}
+	}
+
+	/**
+	 * Updates {@link editType} and {@link editingComponent} based on the option selected from the dropdown
+	 * If the option selected is null or undefined it will automatically send the message to change its value
+	 * and will only display the dropdown with either "null" or "undefined"
+	 * @param data - The option selected from the dropdown
+	 */
+	const onOptionSelect: DropdownProps["onOptionSelect"] = (event, data) => {
+		if (data.optionText === undefined) {
+			throw new Error("Invalid option text from dropdown");
+		}
+
+		if (data.optionText === "undefined") {
+			submitChange(undefined);
+		}
+
+		if (data.optionText === "null") {
+			// We need to support users waiting to use "null" as a value
+			// eslint-disable-next-line unicorn/no-null
+			submitChange(null);
+		}
+
+		let newValue: EditData;
+		if (editProp !== undefined) {
+			newValue = data.optionText === editProp.type ? editProp.value : "";
+		} else {
+			newValue = data.optionText === typeof node.value ? node.value : "";
+		}
+		setEditProp({
+			type: data.optionText,
+			value: newValue,
+		});
+	};
+
+	const options = node.editProps?.editTypes === undefined ? allEdits : node.editProps?.editTypes;
+
+	function getEditType(): string {
+		if (editProp === undefined) {
+			if (node.value === null) {
```

Comment:
Nit: you can simplify this as:
```suggestion
			return node.value === null ? "null" : typeof node.value;
```

Reply:
But I would also extract this into a helper function. Something like:
```typescript
getTypeString(value: unknown): string {
	return value === null ? "null" : typeof value;
}
```